File: android/app/src/main/java/com/tappyai/app/ProductFlags.kt
package com.tappyai.app

/**
 * Product visibility flags — the Android mirror of the web's `src/lib/config/product.ts`.
 *
 * 🚨 A FLAG THAT EXISTS ON ONE PLATFORM ONLY IS THE DEFECT, NOT THE FEATURE. The web file records
 * what `SHOW_SCAM_SHIELD` cost: it gated Android and was read by no web surface at all, so one
 * boolean meant two different products. Anything listed here is flipped on **every** platform in
 * the same change.
 *
 * Android has carried single-use copies of this idea before (`SHOW_PRO_UPGRADE` as a private
 * constant inside `ProfileScreen.kt`). Those stay where they are; this object exists for the flags
 * that more than one package has to agree on.
 */
internal object ProductFlags {
    /**
     * Music — HIDDEN ON EVERY PLATFORM, NOT DELETED.
     *
     * The catalogue served in production is Jamendo API hotlinks and that licensing question is
     * still open (`docs/uat/PHASE7-AUDIT.md` R5), so a user must not be able to reach Music through
     * normal product navigation.
     *
     * 🚨 ON ANDROID THERE IS NOTHING TO GATE (uat/unified, "Huong 2", 2026-09-24). This build took
     * phase7's Android, which had already REMOVED the Android music surface entirely — no
     * `com.tappyai.app.music` package, no registry entry, no composer sound picker, no feed sound
     * pill, no music routes. Music is ABSENT on Android (the strongest hide), and this constant is
     * kept only as the cross-platform MIRROR so the contract stays symmetric. (Web keeps its
     * `/music` LIBRARY behind the flag; Android does not carry it.)
     *
     * Mirrors web `SHOW_MUSIC` (also served as `flags.showMusic` by `GET /api/config`) and iOS
     * `ProductFlags.showMusic`. Flip all three together if Music is ever launched — on Android that
     * also means re-adding the music package, which was removed here.
     */
    const val SHOW_MUSIC = false
}

File: ios/TappyAI/Core/Config/ProductFlags.swift
import Foundation

/// Product visibility flags — the iOS mirror of the web's `src/lib/config/product.ts`.
///
/// 🚨 A FLAG THAT EXISTS ON ONE PLATFORM ONLY IS THE DEFECT, NOT THE FEATURE. The web file records
/// what `SHOW_SCAM_SHIELD` cost: it gated native and was read by no web surface at all, so one
/// boolean meant two different products. Anything listed here is flipped on **every** platform in
/// the same change.
///
/// These are compile-time constants, deliberately. `/api/config` also carries the same values under
/// `flags`, and `ProfileMainView` already reads two of them at runtime — but a runtime flag cannot
/// hide a surface before the first config response lands, and a feature withdrawn for a legal
/// reason must be absent from the first frame. When the decision behind one of these is settled,
/// the constant is the single place to change.
enum ProductFlags {
    /// Music — HIDDEN ON EVERY PLATFORM, NOT DELETED.
    ///
    /// The catalogue served in production is Jamendo API hotlinks and that licensing question is
    /// still open (`docs/uat/PHASE7-AUDIT.md` R5), so a user must not be able to reach Music
    /// through normal product navigation. Nothing underneath is touched — `Features/Music`, the
    /// composer's `MusicPickerView`, `SoundPageView` and their services all stay exactly where they
    /// are, and flipping this one boolean back to `true` restores every iOS entry point at once.
    ///
    /// Mirrors web `SHOW_MUSIC` (served as `flags.showMusic`) and Android
    /// `com.tappyai.app.ProductFlags.SHOW_MUSIC`. Flip all three together.
    static let showMusic = false
}
